### ISSUE TITLE
💅 Rename `noSemicolonInJsx` rule?

### ISSUE THREAD

**Issue by nstepien** [biomejs/biome#2217]
### Environment information

```bash
irrelevant
```

### Rule name

noSemicolonInJsx

### Playground link

https://biomejs.dev/playground/?code=LwAvACAAaABhAGQAIAB0AG8AIABmAGkAbABsACAAdABoAGUAIAAiAFAAbABhAHkAZwByAG8AdQBuAGQAIABsAGkAbgBrACIAIABmAGkAZQBsAGQAIAB3AGkAdABoACAAcwBvAG0AZQB0AGgAaQBuAGcALgAuAC4A

### Expected result

The rule does not disallow all semicolons in JSX, only suspicious ones, so I find the name inadequate.
I had originally [proposed](https://github.com/biomejs/biome/issues/927#issuecomment-1829602600) "`noSuspiciousSemicolonInJSX`", which is better IMO. I'm open to alternatives.
I do realize that it'll end up under the `suspicious` group though, so maybe that's why it was named `noSemicolonInJsx`? Still not all semicolons in jsx are suspicious.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
@fujiyamaorange would you like to help? Unfortunately we didn't catch the name of the name early.

**Comment by nstepien** [comment]
Thankfully it's still in nursery 🙂

**Comment by fujiyamaorange** [comment]
@nstepien 
Thank you for the comment.
As you mentioned, 
>I do realize that it'll end up under the suspicious group though, so maybe that's why it was named noSemicolonInJsx?

this is the reason why I named it `noSemicolonInJsx`.

But if everyone is confused the behavior, I will rename it. 

@ematipico 
What do you think about it?

**Comment by ematipico** [comment]
I think rule names should be chosen independently from their group.

It's possible that in V2 we will move groups around, so the name of a rule should be self-explanatory

**Comment by fujiyamaorange** [comment]
@ematipico 
I got it! I'll rename!
